deepspeed/runtime/engine.py

```diff
@@ -689,6 +689,8 @@ def _do_args_sanity_check(self, args):
                 assert args.deepspeed_config is None, "Not sure how to proceed, we were given both a deepscale_config and deepspeed_config"
             args.deepspeed_config = args.deepscale_config
 
+        assert self.client_optimizer is not None or self.client_model_parameters is not None, "At least one of the optimizer and model parameters must be input to train the model."
```

Comment:
1. Add assertion to DS engine. see https://github.com/microsoft/DeepSpeed/issues/1034